dsl/pace/dsl/dace/orchestration.py

```diff
@@ -55,9 +62,7 @@ def _download_results_from_dace(
     gt4py_results = None
     if dace_result is not None:
         for arg in args:
-            if isinstance(arg, gt4py.storage.Storage) and hasattr(
-                arg, "_set_device_modified"
-            ):
+            if isinstance(arg, cp.ndarray) and hasattr(arg, "_set_device_modified"):
```

Comment:
Also wondering here what happens to Quantity.

Can you use a `try: except` block that calls this method instead of isinstance checks?